File: frontend/src/types/round.ts
// Mirrors backend/models/round.py — keep in sync when fields change

export type ParticipantKind = "human" | "actor";
export type EntryStatus = "awaiting" | "submitted" | "passed";
export type RoundStatus = "idle" | "collecting" | "awaiting_checks" | "resolving" | "resolved";

export interface RoundSubmitRequest {
  adventure_id: string;
  encounter_id?: string | null;
  character_id: string;
  player_text?: string | null;
  passed?: boolean;
}

export interface RoundSubmitResponse {
  encounter_id: string;
  round_number: number;
  resolved: boolean;
  narrative: string | null;
}

export interface RoundEntryView {
  character_id: string;
  character_name: string;
  kind: ParticipantKind;
  status: EntryStatus;
}

export interface RoundStatusResponse {
  encounter_id: string;
  round_number: number;
  status: RoundStatus;
  entries: RoundEntryView[];
  narrative: string | null;
  resolved_at: string | null;
}

export type CheckStatus = "pending" | "resolved";

export interface PendingCheck {
  id: string;
  encounter_id: string;
  round_number: number;
  character_id: string;
  character_name: string;
  skill_key: string;
  skill_name: string;
  minigame_id: string;
  show_target: boolean;
  target: number | null;
  adv_disadv: number;
  die_size: number;
  status: CheckStatus;
  raw_result: Record<string, unknown> | null;
  score: number | null;
}

export interface ResolveCheckResponse {
  check_id: string;
  score: number;
  resolved_round: boolean;
  narrative: string | null;
}

File: backend/models/round.py
from typing import Literal
from pydantic import BaseModel, Field
from .shared import new_id

ParticipantKind = Literal["human", "actor"]
EntryStatus = Literal["awaiting", "submitted", "passed"]
RoundStatus = Literal["collecting", "awaiting_checks", "resolving", "resolved"]


class RoundEntry(BaseModel):
    character_id: str
    character_name: str = "Unknown"
    kind: ParticipantKind
    actor_id: str | None = None
    status: EntryStatus = "awaiting"
    text: str | None = None
    submitted_at: str | None = None


class PendingRound(BaseModel):
    id: str = Field(default_factory=new_id)
    encounter_id: str
    adventure_id: str
    round_number: int = 1
    status: RoundStatus = "collecting"
    entries: list[RoundEntry] = Field(default_factory=list)
    created_at: str
    resolved_at: str | None = None
    narrative: str | None = None


CheckStatus = Literal["pending", "resolved"]


class PendingCheck(BaseModel):
    id: str = Field(default_factory=new_id)
    encounter_id: str
    round_number: int
    character_id: str
    character_name: str
    skill_key: str
    skill_name: str
    minigame_id: str = "dice-roll"
    show_target: bool = True
    target: float | None = None
    adv_disadv: int = 0
    die_size: int = 20
    status: CheckStatus = "pending"
    raw_result: dict | None = None
    score: float | None = None
